### PR TITLE
fix: expose EntryPlugin creation

### LINKED ISSUE #11634: Expose 'entryOption' hooks logic in EntryOptionPlugin
Is it possbile to expose these logic in

https://github.com/webpack/webpack/blob/37a676810c70f87bb282103ef3e05e2160040e02/lib/EntryOptionPlugin.js#L19-L35

Currently, I have to repeat the exact same logic for my childCompiler because the following code does not perform any compilation.

```js
new EntryOptionPlugin().apply(childCompiler);
childCompiler.hooks.entryOption.call(options.context, options.entry);
```

The reason of not having a compilation is the following
https://github.com/webpack/webpack/blob/37a676810c70f87bb282103ef3e05e2160040e02/lib/Compiler.js#L872-L888

Also see https://github.com/webpack/webpack/issues/9240#issuecomment-507092856

### PR DESCRIPTION
**Pull request by wood1986**
fix #11634

### CODE REVIEW

**Review by sokra — CHANGES_REQUESTED** [review]
Code is fine, but I need a test cases to cover the case of using this with a child compiler.

**Comment by wood1986** [comment]
Thanks @sokra, I will take care the test case

File: lib/index.js [hunk 1]
```diff
@@ -137,6 +137,12 @@ module.exports = mergeExports(fn, {
 	get DllReferencePlugin() {
 		return require("./DllReferencePlugin");
 	},
+	get DynamicEntryPlugin() {
+		return require("./DynamicEntryPlugin");
```
Comment (wood1986) [inline lib/index.js:141]: Why do you need to expose this?

File: types.d.ts [hunk 2]
```diff
@@ -10295,6 +10327,8 @@ declare namespace exports {
 		Dependency,
 		DllPlugin,
 		DllReferencePlugin,
+		DynamicEntryPlugin,
```
Comment (wood1986) [inline types.d.ts:10330]: Same there?

File: lib/index.js (hunk 1 above)
Comment (sokra) [reply lib/index.js:141]: It's not really needed for this PR, but missing in general. `EntryPlugin` is exposed, so should be `DynamicEntryPlugin`, which is the counterpart for function entries.

File: test/statsCases/child-compiler-apply-entry-option/webpack.config.js [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+		filename: "[name].js"
+	},
+	plugins: [
+		new TestApplyEntryOptionPlugin({
+			entry: {
+				child: "./child"
+			}
+		})
+	]
+};
```
Comment (sokra) [inline test/statsCases/child-compiler-apply-entry-option/webpack.config.js:20]: Maybe add `stats: { children: true, entrypoints: true }` here to see the child compilation stats too

**Review by sokra — CHANGES_REQUESTED** [review]

**Comment by wood1986** [comment]
I do not know how to get them all green

**Review by sokra — APPROVED** [review]

**Pull request merged**